code/datums/quirks/negative.dm

```diff
@@ -447,14 +449,37 @@
 		if(BODY_ZONE_R_LEG)
 			prosthetic = new/obj/item/bodypart/r_leg/robot/surplus(quirk_holder)
 			slot_string = "right leg"
-	prosthetic.replace_limb(human_holder)
+	prosthetic.replace_limb(human_holder, special = TRUE)
 	qdel(old_part)
-	human_holder.regenerate_icons()
 
 /datum/quirk/prosthetic_limb/post_add()
 	to_chat(quirk_holder, "<span class='boldannounce'>Your [slot_string] has been replaced with a surplus prosthetic. It is fragile and will easily come apart under duress. Additionally, \
 	you need to use a welding tool and cables to repair it, instead of bruise packs and ointment.</span>")
 
+/datum/quirk/quadruple_amputee
+	name = "Quadruple Amputee"
+	desc = "Oops! All Prosthetics! Due to some truly cruel cosmic punishment, all your limbs have been taken from you."
+	icon = "tg-prosthetic-full"
+	value = -8
+	medical_record_text = "During physical examination, patient was found to have all prosthetic limbs"
+	hardcore_value = 6
+
+/datum/quirk/quadruple_amputee/add_unique()
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/list/zones = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	var/list/prosthetics = list(/obj/item/bodypart/l_arm/robot/surplus, /obj/item/bodypart/r_arm/robot/surplus, /obj/item/bodypart/l_leg/robot/surplus, /obj/item/bodypart/r_leg/robot/surplus)
+	for(var/i in 1 to 4)
+		var/path = prosthetics[i]
+		var/obj/item/bodypart/prosthetic = new path
+		var/obj/item/bodypart/old_limbs = human_holder.get_bodypart(zones[i])
+		prosthetic.replace_limb(human_holder, special = TRUE)
+		qdel(old_limbs)
```

Comment:
I don't see this at all chief

- replace_body_part isn't an existing proc, replace_limb is but doesn't qdel the old limb
- if it doesn't matter whether the old limb is qdel'd we should just use replace_limb instead
- otherwise you're asking for a new proc to be created that will be confusing to people getting started with limb code and will only be used in like 1 or 2 places

Reply:
updated